### ISSUE TITLE
Gmail mailto links from Cinnamon notification popup

### ISSUE THREAD

**Issue by jdave23** [linuxmint/Cinnamon#3963]
Hi -

Using Cinnamon 2.4.5 on Ubuntu 14.04 (although this is consistent for previous versions of Cinnamon).

My default email client is Evolution, and it's set as Gnome and Cinnamon preferred handler for mailto links.  Also, xdg-email and all other desktop settings (dconf, gconf) point to Evolution, including mail-notification applet.

When new email arrives from my Gmail account, mail-notification opens Evolution when clicked, but the Cinnamon panel notification applet always opens a browser.  If Chrome is installed the popup "Open" button opens Chrome; remove Chrome and it opens Firefox to the Gmail login.

I've been examining the notification and popupmenu javascript files, trying to trace the origin of this behavior.  Is there any way to get Cinnamon notifications/popups to respect Evolution as my default email user agent, or is this the result of some unrelated setting?

**Comment by ManIVIctorious** [comment]
I have a similar problem with nemo and nautilus. Everything is handled by nemo, except of Dropbox. Every time i use the "open Dropbox folder " functionality it opens nautilus

**Comment by jdave23** [comment]
Yeah, I tried tracing the event with Looking Glass but that didn't help.  Somewhere hidden in the bowels of Cinnamon is a line of code indicating how the notification events are handled, but I can't find it ...

**Comment by dalcde** [comment]
Are you saying that the popup notification (at the top right corner) behaves differently from the notification in the panel notification tray? Also does the button appear as an actual button or a url? It seems to me that the application emitting the notification is responsible for handling the button press events.

**Comment by jdave23** [comment]
Both the popup (which provides a button labelled "Open", not a URL) and the notification tray item behave the same way.

Unfortunately the application emitting the notification is ... ambiguous.  I'm using the GNOME Mail Notification app, which appears in the system tray and is configured to launch Evolution.  The Cinnamon notification (and corresponding popups) launch a web browser with the Gmail URL instead.

**Comment by Vahan86** [comment]
@jdave23, is this issue still relevant to Cinnamon 3?

**Comment by jdave23** [comment]
Not sure -- I upgraded to Ubuntu 16.04, but am running the default cinnamon 2.8.6.  Still an issue here, though.

**Comment by Vahan86** [comment]
@jdave23, is this still an issue for you with Cinnamon 3.6?

**Comment by jdave23** [comment]
Currently on 3.4.6 (Ubuntu Artful), and yes -- notifications still ignore my default mail client.  Will check again when 4.18 LTS comes out.
